unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/MeshCollider/MeshColliderHandler.cs

```diff
@@ -78,13 +78,25 @@ private void CreateCollider(PBMeshCollider model)
                     box.size = new UnityEngine.Vector3(1, 1, 0.01f);
                     break;
                 case PBMeshCollider.MeshOneofCase.Sphere:
-                    SphereCollider sphere = colliderGameObject.AddComponent<SphereCollider>();
-                    collider = sphere;
-                    sphere.radius = 1;
+                {
+                    // since unity's `SphereCollider` can't be deformed using it parent transform scaling
+                    // we a sphere mesh to set as collider
+                    MeshCollider meshCollider = colliderGameObject.AddComponent<MeshCollider>();
+                    collider = meshCollider;
+
+                    primitiveMeshPromise = new AssetPromise_PrimitiveMesh(
+                        AssetPromise_PrimitiveMesh_Model.CreateSphere());
```

Comment:
Unity's sphere is too tessellated for this usage: 
![image](https://user-images.githubusercontent.com/50231608/231129881-88d0589e-16f0-4800-8355-cc5853b0521a.png)


I'd recommend creating a sphere in any 3d software faceted enough to allow precise collision.

Reply:
changed the code to create a sphere collider with less amount of faces.
I think I can't get a small number without sacrificing precise collision, WDYT?
![Screen Shot 2023-04-11 at 12 07 17](https://user-images.githubusercontent.com/4195708/231214376-0b9af75e-f693-4675-9e8f-d3a636cd8dcd.png)